### ISSUE TITLE
board files built with wrong structure

### ISSUE THREAD

**Issue by jrvjrv** [vasl-developers/vasl#1876]
At least three of the board files from this update (https://github.com/vasl-developers/vasl/commit/6900d9043246ec57252da349f2291f5e5c948f78) were built with the wrong .zip structure. They have an extra level of directory instead of having the files at the root of the.zip

**Comment by geezer09** [comment]
which ones?

**Comment by jrvjrv** [comment]
> which ones?

From my brief look, all of them. I looked at 13a, 13b & 2b. I did not look at the remainder. My assumption is that they are all bad.

**Comment by geezer09** [comment]
Nice catch, I fixed this issue and updated our board scripts so it will detect faulty file structures in the future. 

A couple notes, anybody that downloaded these boards (10b, 11a, 13a, 13b, 2b, and 7b) while these faulty zip files were in the repo will need to deleted them from their board directory to redownload the correct version. 

Also @derimmer, while fixing this issue I noticed when downloading boards you are downloading from the develop branch, this means that when pushing to the develop branch and making changes to the board files, you are affecting anybody trying to download or update those boards. I am assuming this is how jrvjrv noticed this issue. I think we should change this behaviour and in the future fetch the boards from the production branch I am working on. Once my workflow is ready this branch should be more stable in terms of board versions as a script will verify this on every commit.

**Comment by derimmer** [comment]
> Also @derimmer, while fixing this issue I noticed when downloading boards you are downloading from the develop branch, this means that when pushing to the develop branch and making changes to the board files, you are affecting anybody trying to download or update those boards. I am assuming this is how jrvjrv noticed this issue. I think we should change this behaviour and in the future fetch the boards from the production branch I am working on. Once my workflow is ready this branch should be more stable in terms of board versions as a script will verify this on every commit.
>
This is a level of sophistication that is beyond my skill set. It sounds completely sensible and I completely agree with your plan.

**Comment by jrvjrv** [comment]
> I am assuming this is how jrvjrv noticed this issue.

I did not notice the issue. It was reported on discord. I verified the report, diagnosed the problem and posted a bug. Although having a separate production branch might be helpful, it probably would not have affected this as there was no verification of the proper structure of the .zip file. The board would have been moved over to production eventually before the bug was discovered.
